analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1124 @@
+  rename(
+    pred_pin_final_fmv_new = pred_pin_final_fmv,
+    pred_pin_initial_fmv_new = pred_pin_initial_fmv
+  ) %>%
+  inner_join(
+    assessment_pin_comparison %>%
+      select(meta_pin, pred_pin_final_fmv, pred_pin_initial_fmv),
+    by = "meta_pin"
+  ) %>%
+  rename(
+    pred_pin_final_fmv_comp = pred_pin_final_fmv,
+    pred_pin_initial_fmv_comp = pred_pin_initial_fmv
+  ) %>%
+  mutate(
+    diff_pred_pin_final_fmv =
+      round(((pred_pin_final_fmv_new - pred_pin_final_fmv_comp) /
+        pred_pin_final_fmv_comp), 4),
+    pred_pin_final_fmv_new = dollar(pred_pin_final_fmv_new),
+    pred_pin_final_fmv_comp = dollar(pred_pin_final_fmv_comp),
+    diff_pred_pin_initial_fmv =
+      round(((pred_pin_initial_fmv_new - pred_pin_initial_fmv_comp) /
+        pred_pin_initial_fmv_comp), 4),
+    pred_pin_initial_fmv_new = dollar(pred_pin_initial_fmv_new),
+    pred_pin_initial_fmv_comp = dollar(pred_pin_initial_fmv_comp)
+  ) %>%
+  inner_join(
+    assessment_data_new %>%
+      distinct(meta_pin, .keep_all = TRUE) %>%
+      select(
+        meta_pin, meta_nbhd_code, loc_longitude,
+        loc_latitude, meta_township_name, {{ target_feature_value }}
+      ),
+    by = "meta_pin"
+  )
+
+# Aggregate to neighborhood level
+if (type == "continuous") {
+  pin_nbhd <- pin_individual %>%
+    group_by(meta_nbhd_code) %>%
+    summarize(
+      !!paste0(target_feature_value, "_neighborhood_mean") :=
+        mean(!!sym(target_feature_value), na.rm = TRUE),
+      !!paste0(target_feature_value, "_neighborhood_median") :=
+        median(!!sym(target_feature_value), na.rm = TRUE),
+      !!paste0(target_feature_value, "_neighborhood_90th") :=
+        quantile(!!sym(target_feature_value), 0.9, na.rm = TRUE)
+    ) %>%
+    inner_join(nbhd, by = c("meta_nbhd_code" = "town_nbhd")) %>%
+    st_as_sf()
+} else {
+  pin_nbhd <- pin_individual %>%
+    group_by(meta_nbhd_code, !!sym(target_feature_value)) %>%
+    count() %>%
+    ungroup() %>%
+    group_by(meta_nbhd_code) %>%
+    mutate(
+      percentage = n / sum(n) * 100
+    ) %>%
+    arrange(meta_nbhd_code, desc(n)) %>%
+    mutate(
+      plurality_factor = first(!!sym(target_feature_value))
+    ) %>%
+    ungroup() %>%
+    select(meta_nbhd_code, !!sym(target_feature_value), percentage, plurality_factor) %>%
+    pivot_wider(
+      names_from = !!sym(target_feature_value),
+      values_from = percentage,
+      names_prefix = "percentage_"
+    ) %>%
+    inner_join(nbhd, by = c("meta_nbhd_code" = "town_nbhd")) %>%
+    st_as_sf()
+}
+
+# Pivot wider for leaflet maps to allow multiple shap values
+leaflet_data <- card_individual %>%
+  select(meta_pin, {{ target_feature_shap }}) %>%
+  group_by(meta_pin) %>%
+  mutate(
+    shap_total = sum(!!sym({{ target_feature_shap }})),
+    variable_index = row_number(),
+    name_col = paste0(
+      deparse(substitute(
+        target_feature_shap
+      )), "_",
+      variable_index
+    )
+  ) %>%
+  pivot_wider(
```

Comment:
In my local drive, I have another pivot wider for the value of the feature, but then it seems like it would just make sense to do card values, rather than pin values, if that is working as the default.